### DISCUSSION TITLE
Add support for specifying SWC's env option in Next.js configuration

### DISCUSSION THREAD

**Discussion by sue71** [vercel/next.js#46979] [Ideas]
### Goals

Allow users to specify SWC's env option

### Non-Goals

n/a

### Background

The original SWC supports options equivalent to babel's preset-env and allows for specifying the `useBuiltIns` equivalent option under `mode`. 

https://swc.rs/docs/configuration/swcrc#supported-browsers

However, Next.js sets SWC options internally, preventing framework users from configuring it directly. This constraint prevents me from safely using polyfills with the `usage` option and migrating to SWC.

### Proposal

There are several options, but for now, I prefer option 2.

### Option 1. Enable specifying SWC options via .swcrc file
While I believe this is the most elegant solution, supporting .swcrc could have some impact, making it difficult to address immediately. There are some related issues on this matter, but no progress has been seen on them yet.

refs
- https://github.com/vercel/next.js/discussions/30413
- https://github.com/vercel/next.js/discussions/30940

### Option 2. Add experimental support for `swcEnv` options and make them configurable in the next.config
Add the `swcEnv` or `swcEnvMode` option as an experimental feature and pass the option to the following code.

https://github.com/vercel/next.js/blob/canary/packages/next/src/build/swc/options.ts#L515
 
This change is minimal and has only a minor impact on existing users.

**Comment by Daniel3711997** [comment]
Any news here?
